bin/proving-service/src/api/prover.rs

```diff
@@ -0,0 +1,199 @@
+use miden_block_prover::LocalBlockProver;
+use miden_objects::{
+    MIN_PROOF_SECURITY_LEVEL, batch::ProposedBatch, block::ProposedBlock,
+    transaction::TransactionWitness, utils::Serializable,
+};
+use miden_tx::{LocalTransactionProver, TransactionProver};
+use miden_tx_batch_prover::LocalBatchProver;
+use tokio::sync::Mutex;
+use tonic::{Request, Response, Status};
+use tracing::instrument;
+
+use crate::{
+    commands::worker::ProverTypeSupport,
+    generated::{ProofType, ProvingRequest, ProvingResponse, api_server::Api as ProverApi},
+    utils::MIDEN_PROVING_SERVICE,
+};
+
+/// A collection of provers.
+///
+/// This struct is used to store the provers for the proving service.
+/// Provers are optional because they are not always enabled.
+struct Provers {
+    tx_prover: Option<LocalTransactionProver>,
+    batch_prover: Option<LocalBatchProver>,
+    block_prover: Option<LocalBlockProver>,
+}
+
+impl Provers {
+    fn new(prover_type_support: ProverTypeSupport) -> Self {
+        let tx_prover = if prover_type_support.supports_transaction() {
+            Some(LocalTransactionProver::default())
+        } else {
+            None
+        };
+
+        let batch_prover = if prover_type_support.supports_batch() {
+            Some(LocalBatchProver::new(MIN_PROOF_SECURITY_LEVEL))
+        } else {
+            None
+        };
+
+        let block_prover = if prover_type_support.supports_block() {
+            Some(LocalBlockProver::new(MIN_PROOF_SECURITY_LEVEL))
+        } else {
+            None
+        };
```

Comment:
Question: should we use `LocalBatchProver::default()` and `LocalBlockProver::default()` to be consistent with how we instantiate the `LocalTransactionProver` on line 31 above?